classes/class-wc-rest-connect-shipping-label-controller.php

```diff
@@ -136,4 +158,140 @@ public function post( $request ) {
+				),
+				200
+			);
+		}
+
+		// Shipping labels should be enabled in account settings.
+		if ( true !== $this->settings_store->get_account_settings()['enabled'] ) {
+			return new WP_REST_Response(
+				array(
+					'is_eligible' => false,
+					'reason'      => 'account_settings_disabled',
+				),
+				200
+			);
+		}
+
+		// Check if the store is eligible for shipping label creation.
+		if ( ! $this->shipping_label->is_store_eligible_for_shipping_label_creation() ) {
+			return new WP_REST_Response(
+				array(
+					'is_eligible' => false,
+					'reason'      => 'store_not_eligible',
+				),
+				200
+			);
+		}
+
+		// If the client cannot create a customs form:
+		// - The store address has to be in the US.
+		// - The origin and destination addresses have to be in the US.
+		$client_can_create_customs_form = isset( $request['can_create_customs_form'] ) ? filter_var( $request['can_create_customs_form'], FILTER_VALIDATE_BOOLEAN ) : true;
+		$store_country                  = wc_get_base_location()['country'];
+		if ( ! $client_can_create_customs_form ) {
+			// The store address has to be in the US.
+			if ( 'US' !== $store_country ) {
+				return new WP_REST_Response(
+					array(
+						'is_eligible' => false,
+						'reason'      => 'store_country_not_supported_when_customs_form_is_not_supported_by_client',
+					),
+					200
+				);
+			}
+
+			// The origin and destination addresses have to be in the US.
+			$origin_address      = $this->settings_store->get_origin_address();
+			$destination_address = $order->get_address( 'shipping' );
+			if ( 'US' !== $origin_address['country'] || 'US' !== $destination_address['country'] ) {
+				return new WP_REST_Response(
+					array(
+						'is_eligible' => false,
+						'reason'      => 'origin_or_destination_country_not_supported_when_customs_form_is_not_supported_by_client',
+					),
+					200
+				);
+			}
+		}
+
+		// If the client cannot create a package (`can_create_package` param is set to `false`), a pre-existing package
+		// is required.
+		$client_can_create_package = isset( $request['can_create_package'] ) ? filter_var( $request['can_create_package'], FILTER_VALIDATE_BOOLEAN ) : true;
+		if ( ! $client_can_create_package ) {
+			if ( empty( $this->settings_store->get_packages() ) && empty( $this->settings_store->get_predefined_packages() ) ) {
+				return new WP_REST_Response(
+					array(
+						'is_eligible' => false,
+						'reason'      => 'no_packages_when_client_cannot_create_package',
+					),
+					200
+				);
+			}
+		}
+
+		// There is at least one non-refunded and shippable product.
+		if ( ! $this->shipping_label->is_order_eligible_for_shipping_label_creation( $order ) ) {
+			return new WP_REST_Response(
+				array(
+					'is_eligible' => false,
+					'reason'      => 'order_not_eligible',
+				),
+				200
+			);
+		}
+
+		// If the client cannot create a payment method (`can_create_payment_method` param is set to `false`), a pre-selected payment method is required.
```

Comment:
Do we need to update "a pre-selected payment method is required." since we allow other payment methods?